components/dashboard/src/usage/download/DownloadUsage.tsx

```diff
@@ -0,0 +1,130 @@
+            return;
+        }
+
+        toast(
+            <DownloadUsageToast
+                orgName={org?.slug ?? org?.id}
+                attributionId={attributionId}
+                startDate={startDate}
+                endDate={endDate}
+            />,
+            {
+                autoHide: false,
+            },
+        );
+    }, [attributionId, endDate, org, startDate, toast]);
+
+    return (
+        <Button type="secondary" onClick={handleDownload} className="flex flex-row" icon={<DownloadIcon />}>
+            Export as CSV
+        </Button>
+    );
+};
+
+type DownloadUsageToastProps = Props & {
+    orgName: string;
+};
+
+const DownloadUsageToast: FC<DownloadUsageToastProps> = ({ attributionId, endDate, startDate, orgName }) => {
+    const [progress, setProgress] = useState(0);
+
+    const queryArgs = useMemo(
+        () => ({
+            orgName,
+            attributionId: AttributionId.render(attributionId),
+            from: startDate.startOf("day").valueOf(),
+            to: endDate.endOf("day").valueOf(),
+            onProgress: setProgress,
+        }),
+        [attributionId, endDate, orgName, startDate],
+    );
+    const { data, error, isLoading, abort, remove } = useDownloadUsageCSV(queryArgs);
+
+    const saveFile = useCallback(() => {
+        if (!data || !data.blob) {
+            return;
+        }
+
+        saveAs(data.blob, data.filename);
+    }, [data]);
+
+    useEffect(() => {
+        return () => {
+            abort();
+            remove();
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    if (isLoading) {
+        return (
+            <div className="space-y-2">
+                <div className="flex flex-row items-center space-x-2">
+                    <span>Preparing usage export</span>
+                </div>
+                <ProgressBar percent={progress} />
+            </div>
+        );
+    }
+
+    if (error) {
+        return (
+            <div>
+                <div className="flex flex-row items-start space-x-2">
+                    <ExclamationIcon className="mt-1 w-4 h-4" />
+                    <span>Error exporting your usage data:</span>
+                </div>
+                <pre className="mt-2 whitespace-normal text-sm">{error.message}</pre>
+            </div>
+        );
+    }
+
+    if (!data || data.blob === null) {
+        return <span>There are no usage records for that date range</span>;
+    }
+
+    return (
+        <div className="space-y-2">
+            <div className="flex flex-row items-center justify-between space-x-2">
+                <div className="flex flex-row items-center space-x-2">
+                    <img src={StatusDoneIcon} className="w-5 h-5" alt="Completed icon" />
+                    <span>Usage export complete.</span>
+                </div>
+                <LinkButton onClick={saveFile} className="text-left text-base">
+                    Download CSV file
```

Comment:
**suggestion**: Taking a closer look at this in the preview environment and thinking it could be better if we limited the action label to one or two words. Similar to the famous Download PDF action, could we make this Download CSV and get away with a shorter and better copy? Thoughts?

```suggestion
                    Download CSV
```

| BEFORE | AFTER |
|--------|--------|
| <img width="471" alt="Frame 1548" src="https://github.com/gitpod-io/gitpod/assets/120486/d373da41-e81f-459f-be52-cc1ebd582be8"> | <img width="471" alt="Frame 1549" src="https://github.com/gitpod-io/gitpod/assets/120486/769ebb9c-06d1-405b-b9a6-f9111764c806"> |